explore/src/main/scala/explore/constraints/ConstraintsSummaryTile.scala

```diff
@@ -103,177 +110,195 @@ object ConstraintsSummaryTile:
+                           ),
+                         column(
+                           MinHAColumnId,
+                           ConstraintGroup.constraintSet.andThen(ConstraintSet.elevationRange).get
+                         )
+                           .setCell(_.value match
+                             case ElevationRange.AirMass(_, _)     => ""
+                             case ElevationRange.HourAngle(min, _) => f"${min.value}%.1f"
+                           )
+                           .sortableBy(_ match
+                             case ElevationRange.AirMass(_, _)     =>
+                               ElevationRange.HourAngle.MinHour - 1
+                             case ElevationRange.HourAngle(min, _) => min.value
+                           ),
+                         column(
+                           MaxHAColumnId,
+                           ConstraintGroup.constraintSet.andThen(ConstraintSet.elevationRange).get
+                         )
+                           .setCell(_.value match
+                             case ElevationRange.AirMass(_, _)     => ""
+                             case ElevationRange.HourAngle(_, max) => f"${max.value}%.1f"
+                           )
+                           .sortableBy(_ match
+                             case ElevationRange.AirMass(_, _)     =>
+                               ElevationRange.HourAngle.MinHour - 1
+                             case ElevationRange.HourAngle(_, max) => max.value
+                           ),
+                         column(CountColumnId, _.obsIds.length),
+                         column(ObservationsColumnId, ConstraintGroup.obsIds.get)
+                           .setCell(cell =>
+                             <.span(
+                               cell.value.toSortedSet.toList
+                                 .map(obsId =>
+                                   <.a(
+                                     ^.href := obsUrl(obsId),
+                                     ^.onClick ==> (_.preventDefaultCB
+                                       >> goToObs(obsId)
+                                       >> props.expandedIds.mod(_ + cell.value)
+                                       >> goToObsSet(ObsIdSet.one(obsId))),
+                                     obsId.toString
+                                   )
+                                 )
+                                 .mkReactFragment(", ")
+                             )
+                           )
+                           .setEnableSorting(false)
+                       )
+          // Memo rows
+          rows  <- useMemo((props.constraintList, props.calibrationObservations)): (cl, calibs) =>
+                     cl
+                       .map: (obsIdSet, cs) =>
+                         (obsIdSet -- calibs)
+                           .map: filtered =>
+                             ConstraintGroup(cs, filtered)
```

Comment:
Why not filter in the caller rather than passing the filtered observations list all the way up to here?

Reply:
> filtered observations list

Do you mean the calibration observations list?

Reply:
> I wonder if there should just be a lazy val in programSummaries that does this?

Sounds good to me

Reply:
This was a good call because it turns out that we don't want the calibration observations in any of the groupings that we calculate in the ProgramSummaries. We were filtering them out way down the chain and not always completely. Which messed up, for example, selections in some of the trees because the calibration obs ids were not being filtered out of the tree representation.